source/Request.cpp

```diff
@@ -39,16 +39,39 @@ namespace HTTP{
 
     bool Request::validate_uri(const std::string& message)
```

Comment:
I found new solution for this function using lambda exp, i have tested it locally and it works, maybe we can use it here 
bool Request::validate_uri_two(const std::string& uri){

        m_uri.from_string(uri);
        bool ret_val = (m_uri.get_scheme().empty())? (m_uri.get_path().empty())? (m_uri.get_authority().empty())? false: true  
                                                                                :true 
                                                    :(m_uri.get_scheme() == http || m_uri.get_scheme() == https)? (m_uri.get_path().empty())? false
                                                                                                                                            :true 
                                                                                                                :false;

        return ret_val;
       
    
    }

Reply:
@AlexandarDjordjevic